src/stories/atoms/typography/heading.tsx

```diff
@@ -1,13 +1,13 @@
 import { FC, ReactNode } from 'react';
 
 export type HeadingProps = {
-	level: 1 | 2 | 3 | 4;
+	tag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "p" | "span" | "div";
 	className?: string;
 	children: ReactNode;
 };
 
-const Heading: FC<HeadingProps> = ({ level, className, children }) => {
-	const Tag = `h${level}` as keyof JSX.IntrinsicElements;
+const Heading: FC<HeadingProps> = ({ tag, className, children }) => {
+	const Tag = `${tag}` as keyof JSX.IntrinsicElements;
```

Comment:
```suggestion
	const Tag = `${tag}` as keyof HTMLElementTagNameMap;
```

as the JSX namespace should be deprecated? `HTMLElementTagNameMap` should be a TS global (if not, might be worth upgrading typescript to latest version)